app/components/views/TransactionPage/index.js

```diff
@@ -0,0 +1,28 @@
+import React, { Component } from "react";
+import { autobind } from "core-decorators";
+import { find } from "lodash";
+import ErrorScreen from "../../ErrorScreen";
+import TransactionPage from "./Page";
+import transactionPageConnector from "../../../connectors/transactionPage";
+
+@autobind
+class Transaction extends Component {
+  render() {
+    return  !this.props.walletService ? <ErrorScreen /> : (
+      <TransactionPage
+        {...{
+          ...this.props,
+          ...this.state,
+          transactionDetails: this.getTransaction(),
+        }}
+      />
+    );
+  }
+
+  getTransaction() {
```

Comment:
If this gets moved to a selector we can probably eliminate this component entirely.

something like:

    viewedTransaction = createSelector(
      [transactions, (state, { params: { txHash }}) => txHash]
      (transactions, txHash) => find(transactions.All, { txHash })
    )

The find pulled into selectors is the functional one though so you will probably use find({ txHash }, transactions.All) instead.